builds/misc/templates/consolidate-edgehub-artifacts.yaml

```diff
@@ -2,24 +2,24 @@
 # EdgeHub's dotnet publish folder will serve as a minimal build context for the image
 parameters:
   dotnet.artifacts.path: 'publish/Microsoft.Azure.Devices.Edge.Hub.Service'
-  mqttd.artifacts.path: 'mqtt/target/'
-  mqttd.config.path: 'mqtt/contrib/edgehub/broker.json'
+  mqtt.artifacts.path: 'mqtt/target/'
+  mqtt.config.path: 'mqtt/contrib/edgehub/broker.json'
   watchdog.artifacts.path: 'edge-hub/watchdog/target'
   all.artifacts.path: 'publish/edge-hub'
 
 steps:
   - bash: |
       # Setup build context
       mkdir -p $(Build.BinariesDirectory)/${{ parameters['all.artifacts.path'] }}
-      mkdir -p $(Build.BinariesDirectory)/${{ parameters['all.artifacts.path'] }}/mqttd
+      mkdir -p $(Build.BinariesDirectory)/${{ parameters['all.artifacts.path'] }}/mqtt
```

Comment:
This directory needed to be changed for consistency. In the perf test build this consolidate script doesn't run. This leads to inconsistencies in the dockerfiles across container build (uses this consolidate template) and perf build (doesn't use).

This inconsistency can be fixed by naming the artifact containing folder "mqtt" vs "mqttd". Therefore in the dockerfiles for all builds we can follow the pattern:
```
ADD ./mqtt/target/[ARCH]/release/mqttd /usr/local/bin/mqttd
```

Reply:
So this isn't totally true anymore since I decided to keep the build context small for the mqtt build. So instead in that dockerfile it is:
```
ADD ./target/[ARCH]/release/mqttd /usr/local/bin/mqttd
```

I am keeping this change because I still think it is cleaner.

Reply:
The build context shrink doesn't work for some reason. After investigation this is caused because post-build, the artifacts folder is missing in the build context. Right now I am thinking this is an azure pipelines problem. I will go back to the larger build context solution that works.

Reply:
Wow I think I figured it out! There is a dockerignore in the mqtt folder which is why the target/ dir disappears when this is the build context. I will remove it.

Reply:
Hm, I'm not sure I understand the problem. I feels to me that removing .dockerignore is not the best thing to do. Maybe we should just copy executable into another folder?

Reply:
Yes that was what I was thinking too after Denis made his comment. For the edgehub build, we have consolidate-edgehub-artifacts template that does similar copying. I will follow the pattern and fix it.

Reply:
Although the consolidate-edgehub-artifacts suffers from the same issue. I will add a task for fixing this under the PBI to decrease the edgehub container build time.